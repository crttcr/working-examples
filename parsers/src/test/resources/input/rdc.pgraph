// RDC is a multi component system developed to capture, catalog, and store messages
// on a Kafka topic at scale.
//
//
// This file is a combination of structural and communication relationships.
// Deployment is not considered.
//
## vertex.auto.id = "GENERATE_SEQUENCE"

(svc: SERVICE
	{
	name = "Raw Data Capture Service"
	}
)

(r: COMPONENT
	{
	id   = "Reader.001",     // Components have identity
	name = "RDC.Reader",
	type = "RDC.Reader",
	}
)
(w: COMPONENT
	{
	id   = "Writer.001",
	name = "RDC.Writer",
	type = "RDC.Writer",
	}
)
(e: COMPONENT
	{
	id   = "EventWriter.001",
	name = "RDC.EventWriter",
	type = "RDC.EventWriter"
	}
)
(t: COMPONENT
	{
	id   = "Terminus.001",
	name = "RDC.Terminus",
	type = "RDC.Terminus"
	}
)

(src: PORT
	{
	label     = "Source",
	direction = "INPUT",
	bindkey   = "Topic.Source",
	type      = "KafkaTopic"
	}
)

(storage: PORT
	{
	label     = "Storage",
	direction = "OUTPUT",
	bindkey   = "AWS.S3",
	type      = "AWS.S3"
	}
)

(events: PORT
	{
	label     = "Events",
	direction = "OUTPUT",
	bindkey   = "Topic.Events",
	type      = "KafkaTopic"
	}
)

(dest: PORT
	{
	label     = "Destination",
	direction = "OUTPUT",
	bindkey   = "Topic.Destination",
	type      = "KafkaTopic"
	}
)

(message_queue: QUEUE : INTERNAL
	{
	label     = "Message Queue"
	}
)

(event_queue: QUEUE : INTERNAL
	{
	label     = "Event Queue"
	}
)

(completion_queue: QUEUE : INTERNAL
	{
	label     = "Completion Queue"
	}
)

// Are these 3 necessary, since the components
// provide more descriptive interaction?  
// Saves having to do graph crawl.
//
(svc)--[:HAS]->(src)
(svc)--[:HAS]->(events)
(svc)--[:HAS]->(dest)

(r)--[:IS_PART_OF]->(svc)
(w)--[:IS_PART_OF]->(svc)
(e)--[:IS_PART_OF]->(svc)
(t)--[:IS_PART_OF]->(svc)

(r)--[:READS_FROM]->(src)
(w)--[:WRITES_TO]->(storage)
(e)--[:WRITES_TO]->(events)

(r)--[:WRITES_TO]->(message_queue)
(w)--[:READS_FROM]->(message_queue)

(w)--[:WRITES_TO]->(event_queue)
(e)--[:READS_FROM]->(event_queue)

(e)--[:WRITES_TO]->(completion_queue)
(t)--[:READS_FROM]->(completion_queue)



