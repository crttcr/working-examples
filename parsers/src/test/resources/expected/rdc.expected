## vertex.auto.id = "GENERATE_SEQUENCE"
VERTEX: svc
	LABEL: SERVICE
		PROP name = "Raw Data Capture Service"
VERTEX: r
	LABEL: COMPONENT
		PROP id = "Reader.001"
		PROP name = "RDC.Reader"
		PROP type = "RDC.Reader"
VERTEX: w
	LABEL: COMPONENT
		PROP id = "Writer.001"
		PROP name = "RDC.Writer"
		PROP type = "RDC.Writer"
VERTEX: e
	LABEL: COMPONENT
		PROP id = "EventWriter.001"
		PROP name = "RDC.EventWriter"
		PROP type = "RDC.EventWriter"
VERTEX: t
	LABEL: COMPONENT
		PROP id = "Terminus.001"
		PROP name = "RDC.Terminus"
		PROP type = "RDC.Terminus"
VERTEX: src
	LABEL: PORT
		PROP label = "Source"
		PROP direction = "INPUT"
		PROP bindkey = "Topic.Source"
		PROP type = "KafkaTopic"
VERTEX: storage
	LABEL: PORT
		PROP label = "Storage"
		PROP direction = "OUTPUT"
		PROP bindkey = "AWS.S3"
		PROP type = "AWS.S3"
VERTEX: events
	LABEL: PORT
		PROP label = "Events"
		PROP direction = "OUTPUT"
		PROP bindkey = "Topic.Events"
		PROP type = "KafkaTopic"
VERTEX: dest
	LABEL: PORT
		PROP label = "Destination"
		PROP direction = "OUTPUT"
		PROP bindkey = "Topic.Destination"
		PROP type = "KafkaTopic"
VERTEX: message_queue
	LABEL: QUEUE
	LABEL: INTERNAL
		PROP label = "Message Queue"
VERTEX: event_queue
	LABEL: QUEUE
	LABEL: INTERNAL
		PROP label = "Event Queue"
VERTEX: completion_queue
	LABEL: QUEUE
	LABEL: INTERNAL
		PROP label = "Completion Queue"
EDGE: HAS
EDGE: HAS
EDGE: HAS
EDGE: IS_PART_OF
EDGE: IS_PART_OF
EDGE: IS_PART_OF
EDGE: IS_PART_OF
EDGE: READS_FROM
EDGE: WRITES_TO
EDGE: WRITES_TO
EDGE: WRITES_TO
EDGE: READS_FROM
EDGE: WRITES_TO
EDGE: READS_FROM
EDGE: WRITES_TO
EDGE: READS_FROM